Modules/_io/fileio.c

```diff
@@ -477,9 +477,31 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
            end of file (otherwise, it might be done only on the
            first write()). */
         PyObject *pos = portable_lseek(self, NULL, 2);
```

Comment:
Wouldn't it be simpler to check for `ESPIPE` in `portable_lseek`. (Perhaps you need to add a parameter?)